packages/type-utils/src/typeFlagUtils.ts

```diff
@@ -1,31 +1,63 @@
 import { unionTypeParts } from 'tsutils';
 import * as ts from 'typescript';
 
+const ANY_OR_UNKNOWN = ts.TypeFlags.Any | ts.TypeFlags.Unknown;
+
 /**
- * Gets all of the type flags in a type, iterating through unions automatically
+ * Gets all of the type flags in a type, iterating through unions automatically.
  */
-export function getTypeFlags(type: ts.Type): ts.TypeFlags {
-  let flags: ts.TypeFlags = 0;
+export function getTypeFlags(type: ts.Type): number | ts.TypeFlags {
+  let flags = 0;
   for (const t of unionTypeParts(type)) {
     flags |= t.flags;
   }
   return flags;
 }
 
+function isFlagSet(flags: number, flag: number): boolean {
+  return (flags & flag) !== 0;
+}
+
+/**
+ * @param flagsToCheck The composition of one or more `ts.SymbolFlags`.
+ */
+export function isSymbolFlagSet(
+  symbol: ts.Symbol,
+  flagsToCheck: number | ts.SymbolFlags,
+): boolean {
+  return isFlagSet(symbol.flags, flagsToCheck);
+}
+
 /**
- * Checks if the given type is (or accepts) the given flags
- * @param isReceiver true if the type is a receiving type (i.e. the type of a called function's parameter)
+ * @param flagsToCheck The composition of one or more `ts.TypeFlags`.
+ * @param isReceiver Whether the type is a receiving type (i.e. the type of a
+ * called function's parameter).
+ * @remarks
+ * Note that if the type is a union, this function will decompose it into the
+ * parts and get the flags of every union constituent. If this is not desired,
+ * use the `isTypeFlagSetSimple` function instead.
  */
 export function isTypeFlagSet(
   type: ts.Type,
-  flagsToCheck: ts.TypeFlags,
+  flagsToCheck: number | ts.TypeFlags,
   isReceiver?: boolean,
 ): boolean {
   const flags = getTypeFlags(type);
 
-  if (isReceiver && flags & (ts.TypeFlags.Any | ts.TypeFlags.Unknown)) {
+  if (isReceiver && isFlagSet(flags, ANY_OR_UNKNOWN)) {
     return true;
   }
 
-  return (flags & flagsToCheck) !== 0;
+  return isFlagSet(flags, flagsToCheck);
+}
+
+/**
+ * Similar to the `isTypeFlagSet` function, but does not decompose unions.
+ * Instead performs a single bit flag check.
+ */
+export function isTypeFlagSetSimple(
+  type: ts.Type,
+  flagsToCheck: number | ts.TypeFlags,
+): boolean {
+  return isFlagSet(type.flags, flagsToCheck);
 }
```

Comment:
* I'm not a fan of words like "simple" in code. Most things aren't simple if you don't know them yet. Calling them that can be intimidating to people who read it and think they should know something they don't.
* This is the same as `tsutils.isTypeFlagSet`. We don't need it.

```suggestion
```